Fixes #8502

saravanak opened the pull request:
This commit fixes the searching of the taxons when searching within a `taxonomy`.

The code is re-arranged a bit to DRY the ransack call.

There is a `skipped` test which is the case of searching for nested taxons. I am leaving that open for discussion in this PR.

saravanak committed:
Fixes #8502

This commit fixes the searching of the taxons when the request searching
within a `taxonnomy`.

The code is re-arranged a bit to DRY the ransack call.

There is a `skipped` test which is the case of searching for nested
taxons. I am leaving that open for discussion in this PR.

bbonislawski commented:
Thank you for your contribution but 19 specs failed and codeclimate reported 4 issues. Please fix these things

saravanak commented:
Sure @bbonislawski . I need some help here: there are 3 containers that are successful runs, while one container fails. 
Can you help me here? I see that the same `build-ci.rb` file has successfully executed in the other 3 containers

bbonislawski commented:
What kind of help do you need? All specs in `spec/controllers/spree/api/v1/taxons_controller_spec.rb` failed because of your changes.

saravanak committed:
Fix code as per rubocop suggestion

saravanak committed:
Use the root taxon id instead of the first child which may or may not be
the root

saravanak committed:
Delete  ignored case for now to make code climate's dup checker happy

saravanak commented:
@bbonislawski I have fixed the commits, just one spacing issue (will run rubocop in local henceforth).. Shoul I squash these commits along with the codeclimate correction?

bbonislawski commented on api/spec/controllers/spree/api/v1/taxons_controller_spec.rb:
```diff
@@ -31,18 +32,31 @@ module Spree
+            api_get :index, taxonomy_id: taxonomy.id, q: { name_cont: name }
+          end
+
+          context 'searching for top level taxon' do
```
You don't test if it searched correctly. This test would pass even if api returned `rust_taxon` as second element in array.

bbonislawski commented:
@saravanak yes, please fix issue with codeclimate, add spec to check if only one item was returned and squash commits. Thanks!

saravanak commented:
I have pushed with the above changes @bbonislawski: 

1. rubocop warning
2. Your comment about the additional case for testing
3. Squashed the commits

 I deleted and recreated the branch, and this lead to the close of the PR. I am not able to reopen this PR now. Please let me know if have to follow a different procedure for squashing. 

Apologies for the trouble if any and thanks for the comments!
